### ISSUE TITLE
Missing ability to add boto3 client config for S3 actions

### ISSUE THREAD

**Issue by victorlin** [awslabs/aws-data-wrangler#532]
I woud like to pass boto3 client config into `wr.s3.to_parquet()` to add [retry config](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/configuration.html#using-the-config-object), but this doesn't seem to be possible.

Observing the code, there is an option to pass `config` in `_utils.client()`:

https://github.com/awslabs/aws-data-wrangler/blob/934601f94bf8c9ecbea9792fb6a521878fb7b503/awswrangler/_utils.py#L86-L88

However, this parameter isn't used by `_S3ObjectBase` (which is then used by [`_fs.open_s3_object()`](https://github.com/awslabs/aws-data-wrangler/blob/934601f94bf8c9ecbea9792fb6a521878fb7b503/awswrangler/s3/_fs.py#L562)):

https://github.com/awslabs/aws-data-wrangler/blob/934601f94bf8c9ecbea9792fb6a521878fb7b503/awswrangler/s3/_fs.py#L226

It would be nice to customize this config at a higher level, since `_fs.open_s3_object()` is used by almost all S3 actions.

**Comment by maxispeicher** [comment]
By default `awswrangler` already enables some retries. The configuration that is used by default is `botocore.config.Config(retries={"max_attempts": 5}, connect_timeout=10, max_pool_connections=10)`.

However if you still want to modifiy I've added some functionality for it on the following branch. If you want to, you can give it a try and set the desired `botocore.config.Config` via the `awswrangler.config.botocore_config` property:
```
pip uninstall awswrangler -y
pip install git+https://github.com/maxispeicher/aws-data-wrangler.git@boto-config-for-s3-actions
```

```python
import awswrangler as wr
import botocore

wr.config.botocore_config = botocore.config.Config(
  retries={"max_attempts": 10}, 
  connect_timeout=20, 
  max_pool_connections=30
)
```

**Comment by igorborgest** [comment]
Released on version [2.4.0](https://github.com/awslabs/aws-data-wrangler/releases/tag/2.4.0).
